vars/checkRequestAssignReleaseOwners.groovy

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+import jenkins.ReleaseMetricsData
+import jenkins.ComponentRepoData
+import java.time.LocalDate
+/**
+ * Library to check and assign release owners to the release issues.
+ * @param Map args = [:] args A map of the following parameters
+ * @param args.inputManifest <required> - Input manifest file(s) eg: [2.0.0/opensearch-2.0.0.yml, 2.0.0/opensearch-dashboards-2.0.0.yml] .
+ * @param args.action <optional> - Action to be performed. Default is 'check'. Acceptable values are 'check', 'request' and 'assign'.
+ */
+void call(Map args = [:]) {
+    def inputManifest = args.inputManifest
+    String action = args.action ?: 'check'
+    def version = extractVersionFromManifest(inputManifest[0])
+    def now = LocalDate.now()
+    def monthYear = String.format("%02d-%d", now.monthValue, now.year)
+    def maintainersIndex = "maintainer-inactivity-${monthYear}"
+
+    // Parameter validation
+    validateParameters(args)
+
+    List<String> componentsMissingReleaseOwners = []
+
+    inputManifest.each { inputManifestFile ->
+        def inputManifestObj = readYaml(file: inputManifestFile)
+        withCredentials([
+                string(credentialsId: 'jenkins-health-metrics-account-number', variable: 'METRICS_HOST_ACCOUNT'),
+                string(credentialsId: 'jenkins-health-metrics-cluster-endpoint', variable: 'METRICS_HOST_URL')]) {
+            withAWS(role: 'OpenSearchJenkinsAccessRole', roleAccount: "${METRICS_HOST_ACCOUNT}", duration: 900, roleSessionName: 'jenkins-session') {
+                def metricsUrl = env.METRICS_HOST_URL
+                def awsAccessKey = env.AWS_ACCESS_KEY_ID
+                def awsSecretKey = env.AWS_SECRET_ACCESS_KEY
+                def awsSessionToken = env.AWS_SESSION_TOKEN
+
+                ReleaseMetricsData releaseMetricsData = new ReleaseMetricsData(metricsUrl, awsAccessKey, awsSecretKey, awsSessionToken, version, this)
+                ComponentRepoData componentRepoData = new ComponentRepoData(metricsUrl, awsAccessKey, awsSecretKey, awsSessionToken, version, maintainersIndex, this)
+
+                inputManifestObj.components.each { component ->
+                    def releaseOwner = releaseMetricsData.getReleaseOwners(component.name)
+                    if (releaseOwner == null || releaseOwner.isEmpty()) {
+                        componentsMissingReleaseOwners.add(component.name)
+                        if (action != 'check') {
+                            handleMissingReleaseOwner(component, releaseMetricsData, componentRepoData, action)
```

Comment:
`notifyMissingReleaseOwner`?

Reply:
It handles both `request` and `assign` actions. Hence used the generic name